### ISSUE TITLE
Small skipping of items during arrow navigation

### ISSUE THREAD

**Issue by silvenon** [tailwindlabs/headlessui#1170]
**What package within Headless UI are you using?**

`@headlessui/react`

**What version of that package are you using?**

v1.5.0

**What browser are you using?**

Chrome

**Reproduction URL**

https://github.com/silvenon/repro/tree/headlessui-combobox-order

- branch `headlessui-combobox-order`
- run `npm run dev`

**Describe your issue**

Biggest problems with skipping were fixed in #994, but I found another small one, hopefully the last one. In a specific case when adding a letter which doesn't remove the currently selected item, but (re)moves some next to it, moving the arrow in that direction skips over the newly added ones, I think. But it doesn't continue skipping, only the first move skips. It's a bit hard to define, maybe you'll understand it better. 😄 

This video is not from the repro, but should be very similar.

https://user-images.githubusercontent.com/471278/155999385-8d621dc4-e680-4409-a3ef-716ae6ab70b1.mov

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

One thing I noticed is that part of your setup code is incorrect:

```js
value={filteredPosts[0]}
```

This will swap out the current value with whatever the first item is when you search. This also means that your input field would be updated every time you set a new value in the Combobox. I would split it up like this:

```js
const [post, setPost] = useState(posts[0] /* or `null` if you don't want a default value */)
// ...
<Combobox
  as="div"
  className="mx-auto max-w-xl transform divide-y divide-gray-100 overflow-hidden rounded-xl bg-white shadow-2xl ring-1 ring-black ring-opacity-5 transition-all"
  value={post}
  onChange={setPost}
>
  {/* ... */}
</Combobox>
```
... note, you can still use the `filteredPosts` variable for actually rendering your items, no worries there :)

---

Back to your bug, thanks for the reproduction, I can reproduce it! The funny part is is that it is related to the original few fixes we did. This should be fixed by #1184, and will be available in the next release.

You can already try it using `npm install @headlessui/react@insiders` or `yarn add @headlessui/react@insiders`.

**Comment by silvenon** [comment]
Thank you so much for super fast responses and fixes, it works now!

> This will swap out the current value with whatever the first item is when you search. This also means that your input field would be updated every time you set a new value in the Combobox.

Thanks for the advice 😄  The effect I was aiming for was for the first item to be _focused_ each time the query changes, however, as you say I misunderstood how the `value` prop worked, mistaking selected for focused. But I'm totally fine not having that.

Happy to have this working, yay! 👌
